### ISSUE TITLE
Destination tensor remains unchanged when compiling index_put_ with boolean mask

### ISSUE THREAD

**Issue by meena-ibm** [torch-spyre/torch-spyre#4870]
### What happened?

In compiled Spyre execution, index_put_ with a boolean mask fails to update any elements selected by the True mask entries. The destination tensor remains unchanged.

### What did you expect to happen?

The elements of the destination tensor corresponding to True entries in the boolean mask should be updated with the corresponding values

### How can we reproduce it?

import torch

DEVICE = "spyre:0"

x = torch.zeros(4, 4, dtype=torch.float32, device=DEVICE)

mask = torch.tensor(
    [
        [True,  False, False, False],
        [False, True,  False, False],
        [False, False, True,  False],
        [False, False, False, True],
    ],
    dtype=torch.bool,
    device=DEVICE,
)

values = torch.tensor(
    [1.0, 2.0, 3.0, 4.0],
    dtype=torch.float32,
    device=DEVICE,
)

def fn(x, mask, values):
    out = x.clone()
    out.index_put_((mask,), values, accumulate=False)
    return out

compiled_fn = torch.compile(fn)

result = compiled_fn(x, mask, values)

print("Result:")
print(result)

expected = torch.zeros(4, 4)
expected[mask.cpu()] = values.cpu()

print("\nExpected:")
print(expected)

print("\nMatch:", torch.allclose(result.cpu(), expected))

### Any environmental details we need to know?

pytorch-2.13

### Relevant log output

```Shell
Result:
tensor([[0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.]], device='spyre:0')

Expected:
tensor([[1., 0., 0., 0.],
        [0., 2., 0., 0.],
        [0., 0., 3., 0.],
        [0., 0., 0., 4.]])
```
